### DISCUSSION TITLE
Listing All Existing Queue and Topic Names

### DISCUSSION THREAD

**Discussion by ongun23** [rabbitmq/rabbitmq-dotnet-client#1365] [General]
Hi,
I was wondering if listing all Queue names or Topic names from the server is possible. In order to scale up and use more consumers I can use multiple queue names for the same process. The below code provides what is the feature but it would be better inside the library.

 `   

       public record QueeViewModel(string name);

        private static async Task<List<QueeViewModel>> GetAllQueueNames(ConnectionFactory factory)
        {
            var httpClient = new HttpClient();

            var byteArray = Encoding.ASCII.GetBytes($"guest:guest");

            httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));

            var response = await httpClient.GetAsync($"http://127.0.0.1:15672/api/queues");

            var responseContent = await response.Content.ReadAsStringAsync();

            return JsonConvert.DeserializeObject<List<QueeViewModel>>(responseContent);
        }

`

**Comment by lukebakken** [comment]
This project will only support the AMQP 0.9.1 protocol.

The following google search ...

https://www.google.com/search?q=C%23+rabbitmq+HTTP+api+client

... leads me to these links:

* https://github.com/thinkco/rabbitmq-httpclient-dotnet
* https://stackoverflow.com/questions/61018490/c-net-api-for-rabbitmq-management-api
